moodle/Sniffs/Files/MoodleInternalSniff.php

```diff
@@ -0,0 +1,116 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Checks that each file contains the standard MOODLE_INTERNAL check
+ *
+ * @package    local_codechecker
+ * @copyright  2016 Dan Poltawski <dan@moodle.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+class moodle_Sniffs_Files_MoodleInternalSniff implements PHP_CodeSniffer_Sniff {
+    /**
+     * Register for open tag (only process onecess per file).
+     */
+    public function register() {
+        return array(T_OPEN_TAG);
+    }
+
+    public function process(PHP_CodeSniffer_File $file, $pointer) {
+        // Special dispensation for behat files.
+        if (basename(dirname($file->getFilename())) === 'behat') {
+            return;
+        }
+
+        // Special dispensation for lang files.
+        if (basename(dirname(dirname($file->getFilename()))) === 'lang') {
```

Comment:
it's curious because for some reason I thought lang files, all them, had the MOODLE_INTERNAL thing. It seems that only 3 of them have it right now:

```
$ ag MOODLE_INTERNAL
competency.php
25:defined('MOODLE_INTERNAL') || die();

grading.php
26:defined('MOODLE_INTERNAL') || die();

plugin.php
26:defined('MOODLE_INTERNAL') || die();
```

Just guessing if we could rule about them not having it, maybe out from this issue scope.